app/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionController.scala

```diff
@@ -151,7 +157,11 @@ class PersonalDetailsCollectionController @Inject()(personalDetailsSubmission: P
 
   def whatIsYourPostCode(completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl]): Action[AnyContent] = Action.async { implicit request =>
     viewConfig.isLoggedIn.flatMap { isLoggedIn: Boolean =>
-       Future.successful(Ok(what_is_your_postcode(postcodeForm, completionUrl, isLoggedIn, failureUrl)))
+      sessionKeyCheck(
+        Future.successful(Ok(what_is_your_postcode(postcodeForm, completionUrl, isLoggedIn, failureUrl))),
+        completionUrl,
+        failureUrl
+      )
     }
   }
 
```

Comment:
can we use your function in this method aswell? it will make it cleaner and will guard against multiple tabs being open (L176 ->182).
we don't need to return BadRequest, we can just use your function OOTB